src/app/app.cpp

```diff
@@ -283,6 +360,56 @@ namespace app
         .install();
     }
 
+    static ccf::TxID txid_from_body(etcdserverpb::GetReceiptRequest&& payload)
+    {
+      auto revision = static_cast<uint64_t>(payload.revision());
+      return ccf::TxID{payload.raft_term(), revision};
+    }
+
+    template <typename In, typename Out>
+    void install_historical_endpoint_with_header_ro(
+      const std::string& package,
+      const std::string& service,
+      const std::string& rpc,
+      const std::string& json_path,
+      const app::grpc::HistoricalGrpcReadOnlyEndpoint<In, Out>& f,
+      ccfapp::AbstractNodeContext& context)
+    {
+      auto grpc_path = fmt::format("/{}.{}/{}", package, service, rpc);
+      auto is_tx_committed =
+        [this](ccf::View view, ccf::SeqNo seqno, std::string& error_reason) {
+          return ccf::historical::is_tx_committed_v2(
+            consensus, view, seqno, error_reason);
+        };
+      make_read_only_endpoint(
+        grpc_path,
```

Comment:
I think that these endpoints should be installed under a different package, for clear separation between the etcd API and this new CCF-specific API. What do you think?